src/common/ram/namespace.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ram
+
+import (
+	"sync"
+)
+
+var (
+	parserFactoriesMu sync.RWMutex
+	parserFactories   = make(map[string]NamespaceParserFactory)
+)
+
+// NamespaceParserFactory factory func for
+type NamespaceParserFactory func() NamespaceParser
+
+// NamespaceParser the namespace parser interface
+type NamespaceParser interface {
+	// Match returns true if resource match the parser
+	Match(resource Resource) bool
+	// Parse returns Namespace exact from the resource
+	Parse(resource Resource) (Namespace, error)
+}
+
+// Namespace the namespace interface
+type Namespace interface {
+	// Kind returns the kind of namespace
+	Kind() string
+	// Resource returns new resource for subresources with the namespace
+	Resource(subresources ...Resource) Resource
+	// Object returns object attached with namespace
+	Object() interface{}
```

Comment:
I don't think the `object()` is a good method name, it's very confusable. Let's use `Identity`

Reply:
`Object` changed to `Identity`